File: Trabajo1/Trabajo1_1.py
'''
Un menú con 3 opciones
    1) Método de Bisección
    2) Método de Falsa Posición
    3) Método de Newton Raphson
'''

decision = 0;


def biseccion():
    print("Entra a la resolucion de biseccion")


def falsa_posicion():
    print("Entra a la resolucion con falsa posicion")


def newton_raphson():
    print("Entra a la resolucion con newton raphson")


print("\nBienvenido usuario a este programa")
print("A continuación se muestra un menu, por favor elige el numero del metodo que deseas utilizar")

print("\t1) Método de Bisección")
print("\t2) Método de Falsa Posición")
print("\t3) Método de Newton Raphson")

while True:
    try:
        decision = int(input("\nIngresa aqui tu seleccion "))
        if (decision < 4 and decision > 0):
            break
        else:
            print("Seleccion no valida dentro del rango")
    except ValueError as VE:
        print(VE)
        print("Seleccion no entera")

print("\nHas seleccionado la opcion {}".format(decision))

if (decision == 1):
    biseccion()
elif (decision == 2):
    falsa_posicion();
else:
    newton_raphson();

File: Trabajo1/Trabajo1_1.java
import java.util.Scanner;

public class Trabajo1_1 {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.println("\n\tBienvenido a este programa");
        System.out.println("Del siguiente menu, por favor selecciona el numero del metodo que quieres elegir");

        System.out.println("\t1) Metodo de Biseccion");
        System.out.println("\t2) Metodo de Falsa Posicion");
        System.out.println("\t3) Metodo de Newton Raphson");

        System.out.print("Ingresa tu seleccion aqui -> ");
        int decision = checaInt(sc);

        switch (decision) {
            case 1:
                biseccion();
                break;
            case 2:
                falsa_posicion();
                break;
            case 3:
                newton_raphson();
                break;
            default:
                System.out.println("Seleccion invalida");
                break;
        }

        sc.close();
    }

    public static void biseccion() {
        System.out.println("Entra a la resolucion con el metodo de biseccion");
    }

    public static void falsa_posicion() {
        System.out.println("Entra a la resolucion con el metodo de falsa posicion");
    }

    public static void newton_raphson() {
        System.out.println("Entra a la resolucion con el metodo de newton raphson");
    }

    public static int checaInt(Scanner sc) {
        int provi = 0;
        while (true) {
            try {
                provi = Integer.parseInt(sc.nextLine());
                if (provi > 0 && provi < 4)
                    return provi;
                else {
                    System.out.println("Tu seleccion no es valida");
                    System.out.print("Ingresa una opcion valida");
                }
                    
            } catch (NumberFormatException e) {
                System.out.println("No fue ingresado un numero entero");
                System.out.print("Ingresa un numero valido: ");
            }
        }

    }
}
